spec/lib/savings-account_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'spec_helper'
+
+describe BankAccounts::SavingsAccount do
+  describe "Class methods:" do
+    it "responds to 'new'" do
+      expect(BankAccounts::SavingsAccount).to respond_to :new
+    end
+
+    describe "inherits methods from Account class" do
+      let(:savings_account) do
+        BankAccounts::SavingsAccount.new("Buster", 50)
+      end
+
+        it "adds deposit(amount) and returns updated balance" do
+          expect(savings_account.deposit(2_000)).to eq(2_050)
+        end
+
+      context "when initial_balance argument < 0" do
+        it "raises an ArgumentError" do
+          expect { BankAccounts::SavingsAccount.new(898776, -100) }.to raise_exception(ArgumentError)
+        end
+      end
+
+      context "when initial_balance argument is nil" do
+        it "raises an ArgumentError" do
+          expect { BankAccounts::SavingsAccount.new(898776, nil) }.to raise_exception(ArgumentError)
+        end
+      end
+    end
+
+    describe "unique responses to inherited methods" do
+      describe "raise_exception when initial_balance is < 10 and > 0" do
+        it "raises an ArgumentError" do
+          expect { BankAccounts::SavingsAccount.new(898776, 5) }.to raise_exception(ArgumentError)
```

Comment:
On thing you can do is test every condition using a loop, instead of relying on one number inside the range.
```
(1...10).each do |initial_deposit|
  expect { BankAccounts::SavingsAccount.new(898776, initial_deposit) }.to raise_exception(ArgumentError)
end
```